File: src/main/java/com/github/simiacryptus/aicoder/openai/LogProbs.java
package com.github.simiacryptus.aicoder.openai;

import com.fasterxml.jackson.databind.node.ObjectNode;

public class LogProbs {
    @SuppressWarnings("unused")
    public CharSequence[] tokens;
    @SuppressWarnings("unused")
    public double[] token_logprobs;
    @SuppressWarnings("unused")
    public ObjectNode[] top_logprobs;
    @SuppressWarnings("unused")
    public int[] text_offset;

    @SuppressWarnings("unused")
    public LogProbs() {
    }

    @SuppressWarnings("unused")
    public LogProbs(CharSequence[] tokens, double[] token_logprobs, ObjectNode[] top_logprobs, int[] text_offset) {
        this.tokens = tokens;
        this.token_logprobs = token_logprobs;
        this.top_logprobs = top_logprobs;
        this.text_offset = text_offset;
    }

}

File: src/main/kotlin/com/github/simiacryptus/openai/LogProbs.kt
package com.github.simiacryptus.openai

import com.fasterxml.jackson.databind.node.ObjectNode

class LogProbs {
    @Suppress("unused")
    var tokens: Array<CharSequence> = arrayOf()

    @Suppress("unused")
    var token_logprobs: DoubleArray = DoubleArray(0)

    @Suppress("unused")
    var top_logprobs: Array<ObjectNode> = arrayOf()

    @Suppress("unused")
    var text_offset: IntArray = IntArray(0)

    @Suppress("unused")
    constructor()

    @Suppress("unused")
    constructor(
        tokens: Array<CharSequence>,
        token_logprobs: DoubleArray,
        top_logprobs: Array<ObjectNode>,
        text_offset: IntArray
    ) {
        this.tokens = tokens
        this.token_logprobs = token_logprobs
        this.top_logprobs = top_logprobs
        this.text_offset = text_offset
    }
}